ghcide/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -137,48 +139,49 @@ runLanguageServer options inH outH getHieDbLoc defaultConfig onConfigurationChan
             registerIdeConfiguration (shakeExtras ide) initConfig
 
             let handleServerException (Left e) = do
-                    logError (ideLogger ide) $
+                    logError logger $
                         T.pack $ "Fatal error in server thread: " <> show e
                     exitClientMsg
                 handleServerException _ = pure ()
+
+                uncaughtError (e :: SomeException) = do
+                    logError logger $ T.pack $
+                        "Unexpected exception on notification, please report!\n" ++
+                        "Exception: " ++ show e
+                    LSP.runLspT env $ LSP.sendNotification SWindowShowMessage $
+                        ShowMessageParams MtError $ T.unlines
+                        [ "Unhandled error, please [report](" <> issueTrackerUrl <> "): "
```

Comment:
I don't believe this is supposed to support markup, it's not a `MarkedString`.

Reply:
VSCode supports some basic markup, see https://github.com/microsoft/vscode/issues/20595#issuecomment-281099832
I expect other clients to support it, and if not, everyone can read markdown :)

Reply:
Ugh, this stupid poorly-specified protocol :roll_eyes: